hal/tests/TESTS/mbed_hal/stack_size_unification/CMakeLists.txt

```diff
@@ -3,9 +3,40 @@
 
 include(mbed_greentea)
 
+if("TARGET_CORTEX_A" IN_LIST MBED_TARGET_DEFINITIONS)
+    set(TEST_SKIPPED "Cortex-A target not supported for this test")
+endif()
+
+macro(GetMemSize resultVar memRegion)
+  # ARGN holds all arguments to function after last named one
+  foreach(ITR ${ARGN})
+    if(ITR MATCHES "${memRegion}*")
```

Comment:
I think we could just use `list(FILTER INCLUDE REGEX ...)` here instead of a `foreach` loop and `if` statement. We do something similar in `mbed_greentea.cmake` to find the baud rate. https://github.com/ARMmbed/mbed-os/blob/4cfbea43cabe86bc3ed7a5287cd464be7a218938/tools/cmake/mbed_greentea.cmake#L135 Maybe it's a bit cleaner this way?